helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -103,7 +103,7 @@ private void refresh(ClusterConfig clusterConfig, InstanceConfig instanceConfig,
    *
    * @param assignableReplica - the replica to be assigned
    */
-  void assign(AssignableReplica assignableReplica) {
+  public void assign(AssignableReplica assignableReplica) {
```

Comment:
Please don't make these methods public. They were designed to be package-private intentionally. The idea is that all the update to the assignment shall be done through ClusterModel interfaces. It will be much easier to ensure the model is consistent.

If you make assign/release public here, what if the caller call the node assign but does not call the context addPartitionToFaultZone? The cluster model will be in a inconsistent state.